### ISSUE TITLE
Increase `quantumespresso.pp` versatility (two step process)

### ISSUE THREAD

**Issue by ramirezfranciscof** [aiidateam/aiida-quantumespresso#655]
I would like to be able to use PP to generate the information for the charge density of individual kpoint-band pairs (using `plotnum=7`, [see description](https://www.quantum-espresso.org/Doc/INPUT_PP.html#idm24)) and then have PP add up a combination of those into a single charge density output (using `filepp(i)` and `weight(i)`, [see description](https://www.quantum-espresso.org/Doc/INPUT_PP.html#idm103)).

If I try to specify both settings of `&INPUTPP` and `&PLOT` at the same time, QE complains:
```
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
     task #         0
     from chdens  : error #         1
     can't mix nfile > 1 with multiple output files
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

It is possible to do this in succession (first run the `&INPUTPP` with `&PLOT` empty and then run the `&PLOT` with `&INPUTPP` empty) but if I try to do this with `quantumespresso.pp` I run into a couple of limitations:

1. `quantumespresso.pp` complains if I don't set up the `INPUTPP.plot_num` (I need this blank for the second step of only doing `&PLOT`).
2. Every step needs to be a different execution of PP and so `quantumespresso.pp` will create a separate folder for it. I can set a `parent_folder` and it will copy the content of the `out` folder, but the outputs of the `&INPUTPP` step are outside and won't be copied in this way.
3. `quantumespresso.pp` also currently requires `PLOT.iflag`, and so it will `&PLOT`-process, retrieve, and parse all the outputs of the first `&INPUTPP` step, which is what I'm trying to avoid.

Am I missing any current way to do this? If not, could we flexibilize a little the previous 3 limitations? I'm sure there are other use cases where one needs to run these two steps separatedly (I understand PP is meant to support this two step approach, as it says in the [description](https://www.quantum-espresso.org/Doc/INPUT_PP.html#idm3)). I could try to do the modifications myself and PR the changes if I get some guidelines on how to best do this.

**Comment by ramirezfranciscof** [comment]
I made a PR with a proof of concept, as you can see linked just above this comment. I explain there what changes I made and why I need them, it should help with the discussion (for general or integral comments we can continue here, but if a question or comments refer to specific lines perhaps you can add to the comment in the PR that explains that change).

**Comment by ramirezfranciscof** [comment]
Ok, this has suddenly dropped in priority since it is no longer a viable path for me to work this way with QE (After implementing this solution I came to notice that I can't leverage it because PP has a hardcoded limit of 7 `filepp`/`weight` inputs).

I still believe this could be useful in some other specific cases, and that eventually QE should remove the arbitrary limit it has hardcoded, so I'll leave both this issue and the PR for discussion.

**Comment by greschd** [comment]
> have PP add up a combination of those into a single charge density output

To solve this specific use case, could you have PP calculate the individual charge densities, and then add them together in Python after `aiida-quantumespresso` has parsed them?

In general, it might still make sense to support running the two steps separately.

**Comment by ramirezfranciscof** [comment]
Yes, this is exactly what I'm doing now. I'm creating a calcjob to wrap around a python script that just adds up all the relevant densities generated by PP. I still had to modify `quantumespresso.pp` so that I can ask it to skip retrieving these densities and parsing them, I may PR that change later when I have time to make it proper =P.

**Comment by greschd** [comment]
Right, that makes sense. If I understood you right, the reason you can't compute the sum of densities to begin with is because you want to choose in a special way which ones are added together, is that correct? But you don't actually need the individual ones, which would be too much disk space wasted?

As an aside: If you're working with `pp.x` a lot, you may be interested in my patches that allow using the `PARENT_FOLDER_SYMLINK` option... but they depend on https://github.com/aiidateam/aiida-core/pull/4727, which is why I haven't opened the corresponding PR yet. If you're running into that bottleneck though (copying between directories on the cluster) I can send you instructions for patching that.

**Comment by ramirezfranciscof** [comment]
Hey @greschd , thanks for the offer/advice! For now the copying for the first PP after the PW doesn't seem to be a bottleneck and thus copying the files seems fine. For my personal calcjob postprocessing I am indeed using the symlink list to avoid having to move all those big files, but this was no big deal because I made that calcjob from scratch, it does not rely on anything QE.

> Right, that makes sense. If I understood you right, the reason you can't compute the sum of densities to begin with is because you want to choose in a special way which ones are added together, is that correct? But you don't actually need the individual ones, which would be too much disk space wasted?

Yes, exactly on the last part. Although I'm not sure what you meat with "the reason you can't compute the sum of densities to begin with is because you want to choose in a special way which ones are added together". Any sum of densities using PP is limited to 7 max, irrespective of wether you are using the weights. Of course you can compute other densities directly using different `plot_num` parameters, but I wouldn't consider that "computing the sum", no?

**Comment by greschd** [comment]
> Of course you can compute other densities directly using different `plot_num` parameters, but I wouldn't consider that "computing the sum", no?

Right, I think we're talking about the same thing here: It's because you need to use `plot_num=7`, and can't directly go to `plot_num=0` (sum of all occupied states).
